src/adapters/__tests__/url.test.js

```diff
@@ -20,6 +20,21 @@ describe('URL', () => {
         urlAdapter.set('bar', 'baz')
         expect(getParams().get('bar')).toBe('baz')
       })
+
+      test('keeps the url hash when setting data', () => {
+        window.history.pushState({}, '', '/url/#hash')
+        urlAdapter.set('foo', 'bar')
+
+        expect(window.location.href).toEqual('http://localhost/url/#hash?foo=bar')
+
+        // We have to reset the window location to prevent other tests from having the /url/#hash as active location
+        window.history.pushState({}, '', '/')
```

Comment:
For this, it would be better to set a [teardown](https://jestjs.io/docs/en/setup-teardown.html) which resets the state after each test.